### PR TITLE
16712 - Custom downloader part redownloading improvement

### PR DESCRIPTION
**Pull request by petrca**

### CODE REVIEW

File: src/Squirrel/DownloadManager.cs [hunk 1]
```diff
@@ -557,15 +559,13 @@ private long GetDownloadedFileSize(string fileUrl)
                 }
                 catch (Exception ex)
                 {
+                    /* PO: Sometimes catches "An existing connection was forcibly closed by the remote host." exception.
+                           This is random exception which may occur while downloading files from the Azure (not tested
```
Comment (ArosPrince) [inline src/Squirrel/DownloadManager.cs]:
```suggestion
                           This is random exception which may occur while downloading files from Azure (not tested
```

File: src/Squirrel/DownloadManager.cs [hunk 2]
```diff
@@ -501,33 +502,34 @@ private long GetDownloadedFileSize(string fileUrl)
...
-                                    return;
-                                }
-
                                 fileStream.Write(buffer, 0, currentBytes);
                                 bytesCounter += currentBytes;
 
                                 lock (filePartsInfo.SyncRoot)
                                 {
-                                    filePartsInfo[index].Progress = (int)((bytesCounter * 100) / (range.End - range.Start));
+                                    filePartsInfo[index].Progress = (int)((bytesCounter * 100) / (range.End - range.Start + 1));
```
Comment (ArosPrince) [inline src/Squirrel/DownloadManager.cs:522]: Can you explain?

**Review by ArosPrince — APPROVED** [review]

File: src/Squirrel/DownloadManager.cs [hunk 3]
```diff
@@ -448,7 +448,7 @@ private long GetDownloadedFileSize(string fileUrl)
             string tempFilePath = "";
 
             // Attempts to download file part.
-            for (int i = 0; i < 3; i++)
+            for (int i = 0; i < 5; i++)
```
Comment (klnmi97) [inline src/Squirrel/DownloadManager.cs]: Would not it be nice to define some CONNECTION_ATTEMPTS constant or config variable?

File: src/Squirrel/DownloadManager.cs [hunk 4]
```diff
@@ -587,6 +587,18 @@ private long GetDownloadedFileSize(string fileUrl)
             }
         }
 
+        /// <summary>
+        /// Gets the size of the file.
+        /// </summary>
+        /// <param name="filename">Name of the file.</param>
```
Comment (klnmi97) [inline src/Squirrel/DownloadManager.cs]: Is it only a file name or path?

Comment (petrca) [reply src/Squirrel/DownloadManager.cs]: It is path and I've changed the param to the "path". I copied it from somewhere. Short explanation why I think there was a filename is that the _FileInfo_ constructor takes "fileName" which is described as fully qualified file name. So, it is also path and it is quite confusing if you don't read the doc.

File: src/Squirrel/DownloadManager.cs (hunk 2 above)
Comment (petrca) [reply src/Squirrel/DownloadManager.cs:522]: Range.End and Range.Start are numbers of bytes indexed from the zero, so you have to +1 to get the file size. I forgot to add it before, however it had minor impact on the progress bar.

File: src/Squirrel/DownloadManager.cs (hunk 3 above)
Comment (petrca) [reply src/Squirrel/DownloadManager.cs]: Would be, agree, changed.

**Review by klnmi97 — APPROVED** [review]

**Pull request merged**